server/game/cards/03_TWI/upgrades/DroidCohort.ts

```diff
@@ -0,0 +1,23 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { UpgradeCard } from '../../../core/card/UpgradeCard';
+
+export default class DroidCohort extends UpgradeCard {
+    protected override getImplementationId() {
+        return {
+            id: '7547538214',
+            internalName: 'droid-cohort',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addGainTriggeredAbilityTargetingAttached({
+            title: 'Create a Battle Droid token.',
+            when: {
+                onCardDefeated: (event, context) => event.card.parentCard === context.source
```

Comment:
This is slightly wrong - it is checking for the defeat of an upgrade whose `parentCard` is the same as this upgrade's. We're giving a "when defeated" ability to the unit itself, so it needs to check for its own defeat.

Actually, there's a helper that should be in place for this that we missed. There is already `UpgradeCard.addGainOnAttackAbilityTargetingAttached()` which would grant an on-attack ability with the right settings, do you think you'd be up for copying that method and creating a new one `addGainWhenDefeatedAbilityTargetingAttached`? All that would need to be changed is the `when` condition.

Reply:
added the function to `UpgradeCard`. I think I got it right. I didn't see any core card tests so I am just using the Droid Cohort for the test.